ui/src/dataSet.ts

```diff
@@ -1,29 +1,55 @@
+import { v4 as uuid } from "uuid";
+
 export class DataSet {
-  constructor() {
-    this.included = new Array<Group>();
+  constructor(underlayName: string, groups: Group[] = []) {
+    this.underlayName = underlayName;
+    this.groups = groups;
+  }
+
+  findGroup(id: string): Group | undefined {
+    return this.groups.find((group) => group.id === id);
   }
 
-  included: Array<Group>;
+  listGroups(kind: GroupKind): Group[] {
+    const groups: Group[] = [];
+    this.groups.forEach((group) => {
```

Comment:
nit: `this.groups.filter` probably better suited

Reply:
Done